apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -2445,6 +2445,121 @@ static void tc_libc_stdio_stdsostream(void)
 }
 
 /**
+* @testcase         tc_libc_stdio_tempnam_tc
+* @brief            Returns a pointer to a string that is a valid filename
+* @scenario         The tempnam() function returns a pointer to a unique temporary filename, or NULL if a unique name cannot be generated.
+* @apicovered       tempnam(), mkstemp(), mktemp()
+* @precondition     NA
+* @postcondition    NA
+*/
+
+static void tc_libc_stdio_tempnam_tc(void)
+{
+	const char filename[] = "tempnam_tc_test";
+	char *ret1 = "";
+	char *ret2 = "";
+	int ret_check = 0;
+	FILE *fp;
+
+	/* TC 1 with valid prefix */
+
+	ret_check = mount(MOUNT_DEV_DIR, CONFIG_LIBC_TMPDIR, "smartfs", 0, NULL);
+	TC_ASSERT_EQ("mount", ret_check, OK);
+
+	ret1 = tempnam(CONFIG_LIBC_TMPDIR, filename);
+	TC_ASSERT_NEQ_CLEANUP("tempnam", ret1, NULL, umount(CONFIG_LIBC_TMPDIR));
+
+	fp = fopen(ret1, "w");
+	TC_ASSERT_NEQ_CLEANUP("fopen", fp, NULL, goto errout1);
+	fclose(fp);
+
+	unlink(ret1);
+	umount(CONFIG_LIBC_TMPDIR);
+
+	/* TC 2 with NULL prefix */
+
+	ret_check = mount(MOUNT_DEV_DIR, CONFIG_LIBC_TMPDIR, "smartfs", 0, NULL);
+	TC_ASSERT_EQ("mount", ret_check, OK);
+
+	ret2 = tempnam(CONFIG_LIBC_TMPDIR, NULL);
+	TC_ASSERT_NEQ_CLEANUP("tempnam", ret2, NULL, umount(CONFIG_LIBC_TMPDIR));
+
+	fp = fopen(ret2, "w");
+	TC_ASSERT_NEQ_CLEANUP("fopen", fp, NULL, goto errout2);
+	fclose(fp);
+
+	unlink(ret2);
+	umount(CONFIG_LIBC_TMPDIR);
+
+	TC_SUCCESS_RESULT();
+
+errout1:
+	unlink(ret1);
+	umount(CONFIG_LIBC_TMPDIR);
+	return;
```

Comment:
instead of  `return`, How about below suggestion and replace direct call to `umount ` with `errout ` in failure case. so that we can reduce code size. 
```
errout2:
    unlink(ret2);
errout1:
    unlink(ret1);
errout:
    umount(CONFIG_LIBC_TMPDIR);
```

Reply:
looks good, have updated the code